tecken/tests/test_sentry.py

```diff
@@ -0,0 +1,229 @@
+                                "self": ANY,
+                                "wrapped_callback": ANY,
+                            },
+                        },
+                        {
+                            "abs_path": "/app/tecken/views.py",
+                            "context_line": ANY,
+                            "filename": "tecken/views.py",
+                            "function": "broken_view",
+                            "in_app": True,
+                            "lineno": ANY,
+                            "module": "tecken.views",
+                            "post_context": [],
+                            "pre_context": ANY,
+                            "vars": {"request": "[Scrubbed]"},
+                        },
+                    ]
+                },
+                "type": "Exception",
+                "value": "Intentional exception",
+            }
+        ]
+    },
+    "extra": {"sys.argv": ANY},
+    "level": "error",
+    "modules": ANY,
+    "platform": "python",
+    "release": "none:unknown",
+    "request": {
+        "data": "",
+        "env": {"SERVER_NAME": "localhost", "SERVER_PORT": "80"},
+        "headers": {
+            "Auth-Token": "[Scrubbed]",
+            "Content-Length": ANY,
+            "Content-Type": "application/x-www-form-urlencoded",
+            "Host": "localhost",
+            "X-Forwarded-For": "[Scrubbed]",
+            "X-Real-Ip": "[Scrubbed]",
+        },
+        "method": "GET",
+        "query_string": "code=%5BScrubbed%5D&state=%5BScrubbed%5D",
+        "url": "http://localhost/__broken__",
+    },
+    "sdk": {
+        "integrations": [
+            "argv",
+            "atexit",
+            "boto3",
+            "dedupe",
+            "django",
+            "excepthook",
+            "logging",
+            "modules",
+            "redis",
+            "stdlib",
+            "threading",
+        ],
+        "name": "sentry.python",
+        "packages": [{"name": "pypi:sentry-sdk", "version": "1.5.12"}],
+        "version": "1.5.12",
+    },
+    "server_name": ANY,
+    "timestamp": ANY,
+    "transaction": "/__broken__",
+}
+
+
+def test_sentry_scrubbing(sentry_helper, transactional_db):
+    """Test sentry scrubbing configuration
+
+    This verifies that the scrubbing configuration is working by using the /__broken__
+    view to trigger an exception that causes Sentry to emit an event for.
+
+    This also helps us know when something has changed when upgrading sentry_sdk that
+    would want us to update our scrubbing code or sentry init options.
+
+    This test will fail whenever we:
+
+    * update sentry_sdk to a new version
+    * update Django to a new version that somehow adjusts the callstack for an
+      exception happening in view code
+
+    In those cases, we should copy the new event, read through it for new problems, and
+    redact the parts that will change using ANY so it passes tests.
+
+    """
+    client = Client(application)
```

Comment:
Here we use the Werkzeug test client instead of the Django one. The Django one is very very fake and exceptions don't trigger Sentry event generation. Using the Werkzeug one treats the whole thing like a regular WSGI application and the "faking" happens as part of the WSGI API so the Sentry events to get generated and everything works splendid.

I spent some time trying to get this working with LiveServerTestCase. That works by spinning off a thread and running a live Django application there that you can connect to using HTTP clients. It does allow Sentry to generate events, but I could only get it to work sometimes for reasons I'm completely unclear on. I suspect there's something about threads and sentry client initialization and the order in which some things happen.